index.js

```diff
@@ -14,16 +22,229 @@ module.exports = postcss.plugin('alter-color', options => {
 	const initialColor = parseColor(options.from);
 	const finalColor = parseColor(options.to);
 
-	console.log(initialColor);
-	console.log(finalColor);
-
 	return (root, result) => {
 		root.walkRules(rule => {
 			rule.walkDecls(decl => {
 				if (canContainColor(decl.prop)) {
 					const parsedValue = cssTree.parse(decl.value, {context: 'value'});
 
-					console.log(JSON.stringify(parsedValue));
+					cssTree.walk(parsedValue, {
+						visit: 'Identifier',
+						enter(node) {
+							if (node.name === initialColor.keyword) {
+								node.name = finalColor.keyword;
+							}
+						}
+					});
+
+					cssTree.walk(parsedValue, {
+						visit: 'HexColor',
+						enter(node) {
+							if (node.value.length > 3) {
+								if (node.value === removeHashSymbol(initialColor.hex)) {
+									node.value = removeHashSymbol(finalColor.hex);
+								}
+							} else {
+								if (node.value === convertToShortHex(removeHashSymbol(initialColor.hex))) {
+									node.value = convertToShortHex(removeHashSymbol(finalColor.hex));
+								}
+							}
+						}
+					});
+
+					cssTree.walk(parsedValue, {
+						visit: 'Function',
+						enter(node) {
+
+							const ast = cssTree.toPlainObject(node);
+
+							const colorGenerate = ast.children.reduce((acc, item) => {
+								return acc += item.value;
+							}, '');
+
+							switch (node.name) {
+								case 'rgb':
+									if(colorGenerate === initialColor.rgb.toString()) {
+										ast.children.map((item, index) => {
```

Comment:
You have many similar block, which can be extracted into one function. For example,

1. You can provide `finalColor.rgb`, `finalColor.rgba`… as a parameter.
2. You can set the number components to process (3 for rgb, 4 for rgba…)

I suggest replace `.map()` with `.forEach()` here. You don’t interested in a result, right? You just want to iterate over array, so `.map()` is irrelevant here.